src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,150 @@ export interface DeleteWorkspaceHookArgs {
 }
 
 
-export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs) : DeleteWorkspaceState => {
+export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs): DeleteWorkspaceState => {
   const [deleting, setDeleting] = useState(false)
   const [loading, setLoading] = useState(false)
-  const [apps, setApps] = useState<DeleteWorkspaceModalLeoApp[]>()
   const [collaboratorEmails, setCollaboratorEmails] = useState<string[] | null>()
   const [workspaceBucketUsageInBytes, setWorkspaceBucketUsageInBytes] = useState<number | null>()
-  const [controlledResourcesExist, setControlledResourcesExist] = useState(false)
-  const [deletableApps, nonDeletableApps] = _.partition(isResourceDeletable('app'), apps) as [DeleteWorkspaceModalLeoApp[], DeleteWorkspaceModalLeoApp[]]
+  const [deletingResources, setDeletingResources] = useState(false)
+  const [workspaceResources, setWorkspaceResources] = useState<WorkspaceResources | null>(null)
```

Comment:
I don't think you have to say "| null" and explicitly say that the initial value is null.

Reply:
Actually, I think I'm wrong about this.

Reply:
@nawatts @cahrens I am fine w/undefined, I will likely want to switch to undefined for all of the other async state as well. Also, I realize this is a prime example of moving to loadedstate which I believe is intended to make this easier to reason about (ticketed but not part of this pr).

Reply:
undefined vs null ties into establishing patterns for TS. Maybe it's worth preferring undefined over null if it makes the type definitions more convenient / ergonomic.

For fetching data from APIs, the ideal is encapsulating that in a custom hook (that probably itself uses useLoadedData). That way, the component doesn't have to deal with anything async and the hook's return value is easily mocked in unit tests.